### ISSUE TITLE
**[Feature]: Add Feature Importance Visualization to the ML Analysis Page**

### ISSUE THREAD

**Issue by dhanasai2** [Arnav-Singh-5080/IPL-Win-Probability-Prediction#404]
** Labels:** `enhancement`, `good first issue`, `GSSoC`, `machine-learning`

---

###  Is your feature request related to a problem? Please describe.

CricScope uses a Logistic Regression model to predict win probabilities, but there is currently no way to see **which features the model actually relies on most** when making a prediction. Users and contributors have no insight into why the model assigns a certain probability — for example, is `runs_left` more important than `crr`? Does `wickets` matter more than `balls_left`?

This is a core ML interpretability gap. Without feature importance, the Analysis page is incomplete from a machine learning standpoint.

---

###  Describe the solution you'd like

Add a **Feature Importance chart** to the existing `Match Analysis` page in `application.py`. 

The Logistic Regression model already exposes `.coef_` after training. The task is to:

1. Extract the feature names from the `ColumnTransformer` preprocessor (both OHE categorical and numeric passthrough)
2. Pair them with the corresponding `LogisticRegression.coef_[0]` values
3. Take the **top 10–15 most influential features** by absolute coefficient value
4. Plot a horizontal bar chart using **Plotly Express** (already imported in the codebase) showing each feature and its coefficient magnitude
5. Style the chart to match the existing dark luxury aesthetic (dark background, gold accent color `#d4af37`)
6. Display it as a new section titled `"Model Feature Weights"` on the Analysis page

---

### Why this is a great beginner ML contribution

- The model is already trained — no retraining needed
- It teaches a fundamental ML concept: **how to interpret linear model coefficients**
- It uses **only libraries already in the project**: scikit-learn, Plotly, Pandas
- The change is **isolated to the Analysis page** — no risk of breaking prediction logic
- It's a visual, satisfying contribution with a clear before/after

---

### 📋 Proposed Implementation Steps

```python
# After pipe is trained, extract feature names and coefficients:

ohe_features = pipe.named_steps['preprocessor'] \
    .named_transformers_['cat'] \
    .get_feature_names_out(['batting_team', 'bowling_team', 'city'])

num_features = ['runs_left', 'balls_left', 'wickets', 'target', 'crr', 'rrr']

all_features = list(ohe_features) + num_features
coefficients = pipe.named_steps['model'].coef_[0]

import pandas as pd
feat_df = pd.DataFrame({'feature': all_features, 'importance': coefficients})
feat_df['abs_importance'] = feat_df['importance'].abs()
top_features = feat_df.nlargest(15, 'abs_importance')

# Then plot with Plotly Express matching the dark gold theme
```

---

### Acceptance Criteria

- [ ] Feature importance chart is visible on the Analysis page
- [ ] Shows at least top 10 features by absolute coefficient value
- [ ] Chart uses dark background with gold/amber accent colors consistent with the existing UI
- [ ] No changes made to the prediction pipeline or model training logic
- [ ] App runs locally with `streamlit run application.py` without errors

---

### Additional Notes

- This feature directly contributes to **ML transparency** and helps users understand the model
- Stick to the **existing Plotly Express** import already in the file — no new dependencies needed
- This is tagged as `good first issue` — if you need guidance on extracting sklearn pipeline feature names, feel free to ask in the comments!

---

**Program:** GSSoC '26   
**Difficulty:** Beginner  
**Area:** Machine Learning / Data Visualization  

---

*I would like to work on this issue. Please assign it to me. I am a GSSoC '26 contributor.*

**Comment by sinchana-aiml** [comment]
Hi @dhanasai2,

I would like to work on this issue under GSSoC'26.

I have reviewed the feature request and understand the goal of adding a Feature Importance visualization using the existing Logistic Regression model coefficients and Plotly. I would be happy to implement the chart while keeping the current prediction pipeline unchanged and maintaining consistency with the existing UI.

Please assign this issue to me if it is available.

Thank you!

**Comment by aditip01-cloud** [comment]
Hii @dhanasai2 

I am a GSSoC'26 contributor and have reviewed the issue. I would like to work on the same.
Please assign me this issue

Thank you

**Comment by sudeepti25** [comment]
Hi , I would like to work on this issue

**Comment by PranavAgarkar07** [comment]
/claim

Will implement: extract feature names from the ColumnTransformer pipeline, pair with LogisticRegression.coef_[0], plot a horizontal bar chart of top 15 features by absolute coefficient value using Plotly Express with the dark gold theme, and add a "Model Feature Weights" section to the Analysis page.

**Comment by Shreya5473** [comment]
Hi, I am a GSSoC 2026 contributor and would like to work on this. I have worked on this before and I would love to complete this issue.

**Comment by ramaarahatekar** [comment]
Hello, this sounds interesting & I am willling to contribute! thanks.

**Comment by ramaarahatekar** [comment]
Implemented the feature and opened a PR for review.

Looking forward to feedback and suggestions for improvement.
